src/sqltrie/sqlite/sqlite.py

```diff
@@ -314,14 +315,10 @@ def ls(
             )
 
     def traverse(self, node_factory, prefix=None):
-        key = prefix or ()
-        row = self._get_node(prefix)
-        value = row["value"]
-
-        children_keys = ((*key, row["name"]) for row in self._get_children(key))
-        children = (self.traverse(node_factory, child) for child in children_keys)
-
-        return node_factory(None, key, children, value)
+        trie = PyGTrie()
+        for key, value in self.items():
+            trie[key] = value
```

Comment:
So this requires reading the whole trie into memory, right? That should become a problem at some size and kinda defeats the purpose a little bit it seems like.

After researching, what do you think about sql-based view as discussed in the previous PR? Just trying to understand if that would be the long term solution.